### PR TITLE
VACMS 14237: fix str replace process plugin

### LINKED ISSUE #14237: Fix str_replace process plugin patch
## Describe the defect
A patch was created as part of https://github.com/department-of-veterans-affairs/va.gov-cms/issues/14148
that changed a noisy Migration stopping failure into a rather silent data blockage for ~10 facilities.
The failure and the scope of it were determined here https://github.com/department-of-veterans-affairs/va.gov-cms/issues/14189

The patch in the upstream issue needs to be reworked as described in this comment upstream
https://www.drupal.org/project/migrate_plus/issues/3368310#comment-15132328

## AC / Expected behavior
- [ ] I expect that when each of our facility migrations are run as update migrations, there are no failures identified.
- [ ] I expect that the data the we save in the CMS from the migration matches the data we stored before 14148 and before the php upgrade that broke the migration in a noisy way.

### PR DESCRIPTION
**Pull request by omahane**
## Description

Closes #14237 

## Testing done
Manually
PHPUnit

#### NCA
- Run the [NCA](https://staging.cms.va.gov/admin/structure/migrate/manage/facility/migrations/va_node_facility_nca/execute) migration (with "Update" checked under "Additional execution options")
 - [x] Validate that the migration had 0 failures
#### Vet Center Outstation
 - Run the [Vet Center Outstation](https://staging.cms.va.gov/admin/structure/migrate/manage/facility/migrations/va_node_facility_vet_centers_os/execute) migration (with "Update" checked under "Additional execution options")
 - [x] Validate that the migration had 0 failures
#### VHA 
 - Open a PHP terminal in [Tugboat](https://tugboat.vfs.va.gov/64ad4fe62d3036648da992d4).
 - Run the following: `drush migrate:import va_node_health_care_local_facility --update --idlist=vha_549QC`
 - [x] Validate that the migration had 0 failures

#### Forms
 - Run the [Forms](https://pr14310-p4eynl3nf4mzsh6rhrhobpefsdmjdvwy.ci.cms.va.gov/admin/structure/migrate/manage/forms/migrations) migration (with "Update" checked under "Additional execution options")
 - [x] Validate that the migration had 0 failures
#### VBA DB
 - Run the VBA DB migration (with "Update" checked under "Additional execution options")
 - [x] Validate that the migration had 0 failures

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `CMS Team`
- [ ] `Public websites`
- [x] `Facilities`
- [ ] `User support`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing announcement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping the UX writer so they are ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping the UX writer to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

**Comment by omahane** [comment]
As a note, the reason we don't need need a combined patch is that the earlier one was for a one-time migration related to COVID, per @swirtSJW:
https://dsva.slack.com/archives/CT4GZBM8F/p1688584436556609?thread_ts=1688582578.256239&cid=CT4GZBM8F

File: tests/phpunit/Migration/VaFacilityNcaMigrationTest.php
Comment (ndouglas) [inline tests/phpunit/Migration/VaFacilityNcaMigrationTest.php:1]: Beautiful test 😍

**Comment by ndouglas** [comment]
@edmund-dunn the links point to staging. Are you sure you ran on Tugboat?

**Comment by ndouglas** [comment]
(asking because I did the same thing earlier 😁 )

**Comment by edmund-dunn** [comment]
Just noticed that! Thanks @ndouglas good catch!

**Comment by ndouglas** [comment]
I ran the NCA and DB migrations and both worked fine. Mentioning it because I know @edmund-dunn is doing others and I don't wanna duplicate effort or being confusing about where checkmarks are coming from 🙂

**Review by ndouglas — APPROVED** [review]

**Pull request merged**